onnxruntime/core/framework/session_state.h

```diff
@@ -175,8 +175,7 @@ class SessionState {
 
   SessionState* GetMutableSubgraphSessionState(onnxruntime::NodeIndex index, const std::string& attribute_name);
 
-  onnxruntime::concurrency::ThreadPool* GetThreadPool() const { return thread_pool_; }
-  void SetThreadPool(onnxruntime::concurrency::ThreadPool* p_pool) { thread_pool_ = p_pool; }
+  concurrency::ThreadPool* GetThreadPool() const { return thread_pool_; }
 
```

Comment:
A const method returning a non-const internal is generally bad. Please have both methods.

Reply:
I know your concern. But I'd rather leave the problem to the next guy. Sorry for that. In this special case, it definitely need more discussion.